control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsHandler.java

```diff
@@ -0,0 +1,108 @@
+package org.oskari.announcements.actions;
+
+import fi.nls.oskari.control.RestActionHandler;
+import org.oskari.announcements.helpers.AnnouncementsParser;
+import org.oskari.announcements.helpers.Announcement;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.log.AuditLog;
+import fi.nls.oskari.util.JSONHelper;
+
+@OskariActionRoute("Announcements")
+public class AnnouncementsHandler extends RestActionHandler {
+    private static final Logger LOG = LogFactory.getLogger(AnnouncementsHandler.class);
+    private AnnouncementsService service = new AnnouncementsServiceMybatisImpl();
+
+    // Handle get announcements
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+
+        try {
+            JSONObject result = new JSONObject();
+            if (params.getUser().isAdmin()) {
+                result = service.getAdminAnnouncements();
+            } else {
+                result = service.getAnnouncements();
+            }
+            System.out.println(result);
+            ResponseHelper.writeResponse(params, 200, result);
+        
+        } catch (Exception e) {
+            LOG.error("Error", e);
```

Comment:
On an exception the frontend doesn't get any response and the message "Error" isn't really helping anyone :) Maybe it should with something like ResponseHelper.writeError()?

Reply:
Or throw an ActionException like you do on update.